lib/Buzz/Middleware/CallbackMiddleware.php

```diff
@@ -37,13 +38,17 @@ public function __construct($callable)
         $this->callable = $callable;
     }
 
-    public function preSend(RequestInterface $request)
+    public function handleRequest(RequestInterface $request, callable $next)
     {
         call_user_func($this->callable, $request);
```

Comment:
this does not allow altering the request anymore, as the PSR-7 one is immutable.

You should allow the callback to return a modified request and use it for the next call (if the return value is `null`, use the original one, or throw if you want to mandate returning a request)

Reply:
This might be a mistake. Thanks!